src/User/src/Service/UserService.php

```diff
@@ -352,40 +352,40 @@ public function sendWelcomeMail(User $user)
      */
     public function updateUser(User $user, array $data = [])
     {
-        if (isset($data['identity']) && !is_null($data['identity'])) {
+        if (!empty($data['identity'])) {
             if ($this->exists($data['identity'], $user->getUuid()->toString())) {
                 throw new ORMException(Message::DUPLICATE_IDENTITY);
             }
             $user->setIdentity($data['identity']);
         }
 
-        if (isset($data['password']) && !is_null($data['password'])) {
+        if (!empty($data['password'])) {
             $user->setPassword(
                 password_hash($data['password'], PASSWORD_DEFAULT)
             );
         }
 
-        if (isset($data['status']) && !empty($data['status'])) {
+        if (!empty($data['status'])) {
             $user->setStatus($data['status']);
         }
 
-        if (isset($data['isDeleted']) && !is_null($data['isDeleted'])) {
+        if (!empty($data['isDeleted'])) {
```

Comment:
Being that the API will probably be used by an admin panel as well, sometimes they might need to "undelete" a user, which is not possible if you execute `$user->setIsDeleted($data['isDeleted']);` only if `!empty($data['isDeleted'])` because: when "undeleting" a user -> means sending `isDeleted=0` -> where 0 is empty so the script does not enter the if condition -> the user.isDeleted remains unchanged.